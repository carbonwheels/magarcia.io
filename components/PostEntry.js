import Link from 'next/link';
import moment from 'moment';
import PropTypes from 'prop-types';
import remark from 'remark';
import remarkReact from 'remark-react';
import MetaInfo from './MetaInfo';
import Content from './Content';
import css from './PostEntry.module.css';

const PostLink = ({ year, month, day, slug, title, intro, readingTime }) => {
  const date = moment(new Date(year, month - 1, day)).format('YYYY/MM/DD');
  return (
    <li className={css.li}>
      <Link
        as={`/${date}/${slug}`}
        href={`/post?id=${slug}&year=${year}&month=${month}&day=${day}`}
      >
        <a className={css.titleLink}>
          <h2>{title}</h2>
        </a>
      </Link>
      <div className={css.summary}>
        <Content small>
          {
            remark()
              .use(remarkReact)
              .processSync(intro).contents
          }
        </Content>
      </div>
      <MetaInfo {...{ year, month, day, readingTime }} small />
    </li>
  );
};

PostLink.propTypes = {
  year: PropTypes.number.isRequired,
  month: PropTypes.number.isRequired,
  day: PropTypes.number.isRequired,
  slug: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  intro: PropTypes.string.isRequired,
  readingTime: PropTypes.string.isRequired
};

export default PostLink;
