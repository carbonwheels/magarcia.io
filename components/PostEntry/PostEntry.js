import Link from 'next/link';
import moment from 'moment';
import MetaInfo from '../MetaInfo';
import Content from '../Content';
import { postShape } from '../../lib/propTypes';
import css from './PostEntry.module.css';
import { buildQuery } from '../../lib/utils';

const PostLink = ({ metadata, slug, title }) => {
  const date = moment(metadata.date);
  const query = {
    id: slug,
    year: date.format('YYYY'),
    month: date.format('MM'),
    day: date.format('DD')
  };
  return (
    <li className={css.li}>
      <Link as={metadata.url} href={`/post?${buildQuery(query)}`}>
        <a className={css.titleLink}>
          <h2>{title}</h2>
        </a>
      </Link>
      <div className={css.summary}>
        <Content>{metadata.intro}</Content>
      </div>
      <MetaInfo {...metadata} small />
    </li>
  );
};

PostLink.propTypes = postShape;

export default PostLink;
