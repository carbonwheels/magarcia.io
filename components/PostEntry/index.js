import PostEntry from './PostEntry';

export default PostEntry;
