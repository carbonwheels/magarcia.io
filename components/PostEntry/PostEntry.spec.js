import React from 'react';
import renderer from 'react-test-renderer';
import PostEntry from './PostEntry';

it('renders correctly', () => {
  const props = {
    id: 'id',
    slug: 'slug',
    title: 'title',
    content: 'content of the file',
    metadata: {
      url: 'http://example.com',
      intro: 'this is the intro',
      date: '2019-13-01',
      layout: 'post',
      tags: ['tag1', 'tag2'],
      readingTime: '5 mins'
    }
  };
  const tree = renderer.create(<PostEntry {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
