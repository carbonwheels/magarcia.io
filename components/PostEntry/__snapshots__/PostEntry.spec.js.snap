// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<li
  className="li"
>
  <a
    className="titleLink"
    href="http://example.com"
    onClick={[Function]}
  >
    <h2>
      title
    </h2>
  </a>
  <div
    className="summary"
  >
    <div
      className="postContent"
    >
      <p>
        this is the intro
      </p>
    </div>
  </div>
  <div
    className="postMetaInline small"
  >
    <time
      dateTime={null}
    >
      Invalid date
    </time>
    <span
      className="middotDivider"
    />
    <span
      className="readingTime"
      title="5 mins"
    />
  </div>
</li>
`;
